src/core/Akka.NodeTestRunner/Program.cs

```diff
@@ -66,15 +91,80 @@ static void Main(string[] args)
                             specFail.FailureExceptionTypes.Add(ex.GetType().ToString());
                             specFail.FailureMessages.Add(ex.Message);
                             specFail.FailureStackTraces.Add(ex.StackTrace);
+                            _logger.Tell(specFail.ToString());
                             Console.WriteLine(specFail);
+
+                            //make sure message is send over the wire
+                            FlushLogMessages();
                             Environment.Exit(1); //signal failure
+                            return 1;
                         }
-                        sink.Finished.WaitOne();
-                        Environment.Exit(sink.Passed ? 0 : 1);
+
+                        var timedOut = false;
+                        if (!sink.Finished.WaitOne(MaxProcessWaitTimeout)) //timed out
```

Comment:
**IMPORTANT**

I suspect we have some issues with deadlocks inside the `TestConductor` inside `Akka.Remote.TestKit` - in order to work around that and not have the MNTR hang the build agent for 60 minutes (that's the timeout specified by FAKE), I set up a timeout here within each local `NodeTestRunner.exe` process.

If any one of these processes takes longer than `MaxProcessWaitTimeout` (1.5 minutes, currently) the process will exit early and report a failure. 

We will likely need to tune this code later, but for the time being I think it's a reasonable safeguard to prevent hangs.